### PR TITLE
add jaffle shop generator tap

### PR DESCRIPTION
**Pull request by pnadolny13**
@aaronsteers let me know what you think. I didnt know what to put for the image so I took a screenshot of their repo title with the emojis, happy to change it to whatever else.

### CODE REVIEW

File: _data/meltano/extractors/tap-jaffle-shop/meltanolabs.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+capabilities:
+- about
+- catalog
+- discover
+- schema-flattening
+- state
+- stream-maps
+description: The Jaffle Shop Generator, aka "jafgen" is a python-based data generated
+  created by dbt Labs.
```
Comment (aaronsteers) [inline _data/meltano/extractors/tap-jaffle-shop/meltanolabs.yml:9]:
```suggestion
description: The Jaffle Shop Generator, aka "jafgen" is a python-based data generator
  created by dbt Labs.
```

File: _data/meltano/extractors/tap-jaffle-shop/meltanolabs.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+- description: Config object for stream maps capability. For more information check
+    out [Stream Maps](https://sdk.meltano.com/en/latest/stream_maps.html).
+  kind: object
+  label: Stream Maps
+  name: stream_maps
+- description: A name prefix to apply to all streams. Note that the dash ('-') character
+    will be interpreted by many targets as a delimiter between schema and table name.
+  kind: string
+  label: Stream Name Prefix
+  name: stream_name_prefix
```
Comment (aaronsteers) [inline _data/meltano/extractors/tap-jaffle-shop/meltanolabs.yml:45]:
```suggestion
  name: stream_name_prefix
  value: "tap_jaffle_shop-raw_"
```

So users will know from looking at the hub that table/stream names are prefixed by "raw_" by default.

File: _data/meltano/extractors/tap-jaffle-shop/meltanolabs.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+- description: The number of years to simulate data for.
+  kind: integer
+  label: Years
+  name: years
+settings_group_validation:
+- - stream_name_prefix
+  - years
+settings_preamble: ''
+usage: ''
```
Comment (aaronsteers) [inline _data/meltano/extractors/tap-jaffle-shop/meltanolabs.yml:54]:
```suggestion
usage: |
  This dataset generator will simulate real-looking data for the following streams:
    - `stores`
    - `customers`
    - `products`
    - `orders`
    - `items`
    - `supplies`

  Note: This tap supports `state` but does not yet support incremental refresh. Each sync
  operation will generate a new dataset.
```

File: _data/meltano/extractors/tap-jaffle-shop/meltanolabs.yml (hunk 2 above)
Comment (pnadolny13) [reply _data/meltano/extractors/tap-jaffle-shop/meltanolabs.yml:45]: @aaronsteers theres `value` and `placeholder`, did you want it to default to that value or just show an example?

**Comment by aaronsteers** [comment]
@pnadolny13 - This is really great. Thank you! A few suggestions inline in the code. (I hope I didn't break formatting. 😅 )

File: _data/meltano/extractors/tap-jaffle-shop/meltanolabs.yml (hunk 2 above)
Comment (aaronsteers) [reply _data/meltano/extractors/tap-jaffle-shop/meltanolabs.yml:45]: Let's default to the value. That's the same value that the tap provides, but this overriding that is fine, and the transparency is preferable to them having to go check the tap for its built-in defaults.

**Review by aaronsteers — COMMENTED** [review]
Thanks so much! Feel free to merge when ready.

Regarding the logo, I think what you have is fine. You could also use an SVG or higher-res version of the Sandwich 🥪 emoji. I recently learned that a "jaffle" is a type of sandwich. 🤯

**Review by aaronsteers — APPROVED** [review]

**Comment by aaronsteers** [comment]
image

In Google Docs, I pasted "sandwich factory" and bumped to 100 point font 😄 

🥪🏭

**Comment by aaronsteers** [comment]
Thanks, @pnadolny13 !!

**Pull request merged**